crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -986,77 +986,25 @@ impl CheckpointStore {
     }
 
     /// Updates the store on the basis of transactions that have been processed. This is idempotent
-    /// and nothing unsafe happens if it is called twice. Returns the lowest checkpoint number with
-    /// unprocessed transactions (this is the low watermark).
+    /// and nothing unsafe happens if it is called twice.
     fn update_processed_transactions(
         &mut self, // We take by &mut to prevent concurrent access.
         transactions: &[(TxSequenceNumber, ExecutionDigests)],
     ) -> Result<(), SuiError> {
-        let in_checkpoint = self
-            .tables
-            .transactions_to_checkpoint
-            .multi_get(transactions.iter().map(|(_, tx)| tx))?;
-
-        let batch = self.tables.transactions_to_checkpoint.batch();
-
-        // Check we are not re-proposing the same transactions that are already in a
```

Comment:
Let's say it's passed twice. Should that block the checkpoint process? My understanding is that our checkpoint creation is actually tolerant of transactions that were included in previous checkpoints (because of the assumption of byzantine nodes).

Reply:
Alternatively we can also try to detect that and filter them out here (and log an error), but still I think it shouldn't block the checkpoint process if somehow we are passing the same tx twice from batch

Reply:
The issue is that we want to treat crash-recover as non byzantine, because it can happen to more than f for legitimate reasons (net halt to update). So it is worth ensuring this works even if all validators do it at the same time.